File: python/leap/leap.py
"""
Module for check is a year is a leap year
"""

def leap_year(year: int) -> bool:
    """
    Check if a year is a leap year
    """
    def is_div(a: int, b: int) -> bool:
        """
        Checks if the first value is is_divisible by second value
        """
        return a % b == 0 if b != 0 else False

    return is_div(year, 4) and not is_div(year, 100) or is_div(year, 400)

File: lua/leap/leap.lua
local is_divisible, leap_year

--- Returns whether the first argument is divisible by the second.
--- @param dividend integer
--- @param divisor integer
--- @return boolean
function is_divisible(dividend, divisor)
    return dividend % divisor == 0
end

--- Returns whether the given year is a leap year or not.
--- @param year integer
--- @return boolean
function leap_year(year)
    return (
        is_divisible(year, 4) and
        not is_divisible(year, 100) or
        is_divisible(year, 400)
    )
end

return leap_year
